client/src/test/java/org/opensearch/ml/client/MachineLearningNodeClientTest.java

```diff
@@ -624,14 +626,54 @@ public void deploy() {
         }).when(client).execute(eq(MLDeployModelAction.INSTANCE), any(), any());
 
         ArgumentCaptor<MLDeployModelResponse> argumentCaptor = ArgumentCaptor.forClass(MLDeployModelResponse.class);
-        machineLearningNodeClient.deploy(modelId, DeployModelActionListener);
+        machineLearningNodeClient.deploy(modelId, deployModelActionListener);
 
         verify(client).execute(eq(MLDeployModelAction.INSTANCE), isA(MLDeployModelRequest.class), any());
-        verify(DeployModelActionListener).onResponse(argumentCaptor.capture());
+        verify(deployModelActionListener).onResponse(argumentCaptor.capture());
         assertEquals(taskId, (argumentCaptor.getValue()).getTaskId());
         assertEquals(status, (argumentCaptor.getValue()).getStatus());
     }
 
+    @Test
+    public void createConnector() {
+
+
+        String connectorId = "connectorId";
+
+        doAnswer(invocation -> {
+            ActionListener<MLCreateConnectorResponse> actionListener = invocation.getArgument(2);
+            MLCreateConnectorResponse output = new MLCreateConnectorResponse(connectorId);
+            actionListener.onResponse(output);
+            return null;
+        }).when(client).execute(eq(MLCreateConnectorAction.INSTANCE), any(), any());
+
+        ArgumentCaptor<MLCreateConnectorResponse> argumentCaptor = ArgumentCaptor.forClass(MLCreateConnectorResponse.class);
+
+        Map<String, String> params = Map.ofEntries(Map.entry("endpoint", "endpoint"), Map.entry("temp", "7"));
+        Map<String, String> credentials = Map.ofEntries(Map.entry("key1", "key1"), Map.entry("key2", "key2"));
+
+        MLCreateConnectorInput mlCreateConnectorInput = MLCreateConnectorInput.builder()
+                .name("test")
+                .description("description")
+                .version("testModelVersion")
+                .protocol("testProtocol")
+                .parameters(params)
+                .credential(credentials)
+                .actions(null)
+                .backendRoles(null)
```

Comment:
Curious if it's better to add these and state "null" or just leave them off the builder?

Reply:
Added temp backend roles.